.github/workflows/gradle_release_docs.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Release JavaDocs
+
+on:
+  release:
+    types: [ released, prereleased ]
+    branches: [ '3.x' ]
+    tags:
+      - 'v3.*.*'   
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    env:
+      # Define secrets at https://github.com/ReactiveX/RxJava/settings/secrets/actions
+      # ------------------------------------------------------------------------------ 
+      JAVADOCS_TOKEN: ${{ secrets.JAVADOCS_TOKEN }}
+      # ------------------------------------------------------------------------------ 
+      CI_BUILD_NUMBER: ${{ github.run_number }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Cache Gradle packages
+      uses: actions/cache@v2
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+        restore-keys: ${{ runner.os }}-gradle
+    - name: Grant execute permission for gradlew
```

Comment:
Can't we do this step locally, make it executable & push the changes? Or is there any other reason this is needed

Reply:
Windows checkouts and pushes tend to break this so it is safer to have the script grant execute rights as it goes.